### ISSUE TITLE
Enabling annotations causes our arrow functions to lose their names.

### ISSUE THREAD

**Issue by rockymeza** [codemix/flow-runtime#65]
Thanks for creating flow-runtime

This is a:

 - [x] Bug Report
 - [x] Feature Request
 - [ ] Question
 - [ ] Other

Which concerns:

 - [ ] flow-runtime
 - [x] babel-plugin-flow-runtime
 - [ ] flow-runtime-validators
 - [ ] flow-runtime-mobx
 - [ ] flow-config-parser
 - [ ] The documentation website

---

### What is the current behaviour?
When using babel-plugin-flow-runtime with annotations enabled, our stateless functional React components lose their display names. Here is an example that you can run in <https://codemix.github.io/flow-runtime/#/try>:

```javascript
type Props = {
  id: string | number;
  name: string;
};

const MyComponent = (props: Props) => {};

console.log(MyComponent.name);
```

with annotations, this gets transformed to:
```javascript
const MyComponent = t.annotate(props => {
  let _propsType2 = Props;
  t.param("props", _propsType2).assert(props);
}, t.function(t.param("props", Props)));
```

If annotations are disabled, the above code logs `'MyComponent'`.
If annotations are enabled, the above code logs `''`.

---

### What is the expected behaviour?

I would expect that we wouldn't lose the `Function.name` when annotations are enabled.

---

I was trying to figure out how this could be accomplished.

Perhaps it could be compiled to something like this:

```javascript
const MyComponent = t.annotate(function MyComponent(props) {
  let _propsType2 = Props;
  t.param("props", _propsType2).assert(props);
}.bind(this), t.function(t.param("props", Props)), "MyComponent");
```

I'm not really sure what the right route is to go with this.

---

### Which package versions are you using?
"babel-plugin-flow-runtime": "0.5.0"

**Comment by phpnode** [comment]
I guess in annotation mode we could expand any annotated arrow function and give it a name, but it will only work in simple cases where it's the right hand side of a simple assignment expression, but I guess that's how ES6 function name resolution works anyway so probably not an issue.

**Comment by phpnode** [comment]
released in 0.7.0

**Comment by rockymeza** [comment]
@phpnode thanks for looking at this so quickly. I did have a comment on https://github.com/codemix/flow-runtime/commit/7d9651297eb6ffa473e8e7eca497b84108314f41#commitcomment-21149187
